docs/R-IOC/r07_data_import_export.md

```diff
@@ -0,0 +1,324 @@
+
+With the functions from the basic package "utils":
+
+```r
+## read a comma separated file ".csv"
+my_csv <- read.csv(
+  file = "path/to/my_file.csv",
+  header = TRUE,    # whether the file has a colnames in the 1st row
+  sep = ",",        # the character used in the file to separate columns
+  quote = "\"",     # the character(s) for quotes
+  dec = ".",        # the character for decimal points
+  fill = TRUE,      # in case of rows with unequal length, whether to add empty field
+  comment.char = "" # character used to indicated rows as comment lines, use empty string to turn off the interpretation of comment lines
+)
+
+## read a tab separated file ".tsv"
+my_delim <- read.delim(
+  file = "path/to/my_file.tsv",
+  header = TRUE,
+  sep = "\t",
+  quote = "\"",
+  dec = ".",
+  fill = TRUE,
+  comment.char = ""
+)
+
+## read a text file ".txt"
+### sometimes read.table does not work well, be careful of the parameter settings
+my_table <- read.table(
+  file = "path/to/my_file.txt",
+  header = TRUE,
+  sep = " ",
+  quote = "\"'",     # the character(s) for quotes
+  dec = ".",
+  comment.char = "#"
+)
+```
+
+There are other useful arguments that we didn't mentionned here,
+such as `na.strings` (characters to be interpreted as `NA` values), `colClasse` (type of columns), *etc.*, 
+please check the document with `?read.table`.
+
+All these functions from `utils` will return a [data.frame](r06_df_matrices.md).
+
+Apart from the `utils` package, we can use other packages for instance 
+<code>[readr](https://cran.r-project.org/web/packages/readr/index.html)</code> and
+<code>[vroom](https://cran.r-project.org/web/packages/vroom/index.html)</code> to achieve the same goal.
+
+
+### Read Excel Files (.xlx, .xlsx)
+
+Excel files are very often used to store clinical and biological experiment data.
+We can use the <code>[readxl](https://cran.r-project.org/web/packages/readxl/index.html)</code>
+to import them into R:
+
+```r
+# install.packages("readxl")
+library("readxl")
+my_xls <- read_xls(
+  path = "path/to/my_file.xls",
+  sheet = NULL,                 # sheet to read, can be the sheet number or the sheet name
+  range = NULL,                 # a cell range to read
+  col_names = TRUE,             # use the 1st row as column names
+  col_types = NULL,             # type of columns, use "NULL" to guess automatically the type of each column
+  na = "",                      # characters to be interpreted as `NA` values
+  trim_ws = TRUE,               # trim surrounding spaces
+  skip = 0,                     # number of rows to skip before reading
+  n_max = Inf,                  # maximum number of rows to read
+  guess_max = min(1000, n_max), # maximum number of rows to use for guessing column type
+  .name_repair = "unique"       # argument passed to tibble::as_tibble, default is to ensure unique and not empty column names
+)
+
+my_xlsx <- read_xlsx(
+  path = "path/to/my_file.xlsx",
+  sheet = NULL,
+  range = NULL,
```

Comment:
Maybe add an example for these parameters in order to see the structure just once here (and make hard exercices afterward). I saw that in the help section that there is multiple ways to write them, so we can go with really specific examples

Reply:
added ^^